### DISCUSSION TITLE
worker.start must async, otherwise msw will slow then the first request

### DISCUSSION THREAD

**Discussion by Fallsleep** [mswjs/msw#2363] [Show and tell]
In my project, the app will get user info if cookie has user token immediately after page loaded. 
After upgrage to 2.x, this request run into backend server instead of msw worker.
I have long time not working with this project, and today I work on it again.

After some try and search, I find if `worker.start` set to async `await`, the request run well:

```typescript
const { worker } = await import('./browser')
👇
await worker.start({
  onUnhandledRequest: 'bypass',
  serviceWorker: {
    // if homepage not '/'
    url: `${import.meta.env.VITE_APP_ROUTER_BASE_URL}/mockServiceWorker.js`,
  },
})
```

**Comment by kettanaito** [comment]
Hi.

`worker.start()` has been async for a long time, even in MSW v1.0. This is extremely clearly state in both docs:

## 2.x

<img width="872" alt="Screenshot 2024-11-27 at 19 03 24" src="https://github.com/user-attachments/assets/4f34fb03-085a-4010-9549-67c8fac94e05">

## 1.x

<img width="762" alt="Screenshot 2024-11-27 at 19 04 27" src="https://github.com/user-attachments/assets/bd4c8d4b-affb-4059-bb07-e4b1a389707d">

> [!IMPORTANT]
> Okay, I think I see the source of confusion. The V1 docs didn't mention `await worker.start()` in any of the usage examples. That's on me. Those docs are readonly now so I cannot update them. Please always `await worker.start()`.
